etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/messagelink/MessageLink.java

```diff
@@ -0,0 +1,53 @@
+package gov.hhs.cdc.trustedintermediary.etor.messagelink;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * This class represents a link between messages. Each link has a unique ID and is associated with a
+ * set of message IDs to link.
+ */
+public final class MessageLink {
```

Comment:
Thoughts on making this a record?

Reply:
Not sure if it makes sense here as we need to manage state for message ids. It's not supposed to be immutable

Reply:
Yes. What remains immutable is the reference to the Set object, but you can modify the contents of the Set itself.